epub33/core/index.html

```diff
@@ -4362,123 +4320,118 @@ <h4>Scripting Contexts</h4>
 							</li>
 						</ul>
-					</aside>
-				</section>
 
-				<section id="sec-scripted-container-constrained">
-					<h4>Container-Constrained Scripts</h4>
+						<p id="confreq-cd-scripted-container">A container-constrained script MUST NOT contain
+							instructions for modifying the DOM of the EPUB Content Document that embeds it (i.e., the
+							one that contains the <code>iframe</code> element). It also MUST NOT contain instructions
+							for manipulating the size of its containing rectangle.</p>
 
-					<p id="confreq-cd-scripted-container">A <a href="#sec-scripted-content-type-container-constrained"
-							>container-constrained script</a> MUST NOT contain instructions for modifying the DOM of the
-						parent Content Document or other contents in the EPUB Publication. It also MUST NOT contain
-						instructions for manipulating the size of its containing rectangle.</p>
-				</section>
+						<p>Authors should note that <a
+								href="https://www.w3.org/TR/epub-rs-33/#sec-scripted-content-rs-reqs">support for
+								container-constrained scripting in Reading Systems</a> is only recommended in reflowable
+							documents [[EPUB-RS-33]]. Furthermore, Reading System support in fixed-layouts EPUBs is
+							optional.</p>
 
-				<section id="sec-scripted-spine">
-					<h4>Spine-Level Scripts</h4>
-
-					<p id="confreq-cd-scripted-spine"><a>Top-level Content Documents</a> that include <a
-							href="#sec-scripted-content-type-spine-level">spine-level scripting</a> SHOULD remain
-						consumable by the user without any information loss or other significant deterioration when
-						scripting is disabled (e.g., by employing progressive enhancement techniques).</p>
+						<p>Ensuring container-constrained scripts degrade gracefully in Reading Systems without
+							scripting support is advised (see <a href="#sec-scripted-fallbacks"></a>).</p>
+					</section>
 
-					<p>Since Reading Systems are not required to support scripting, failing to account for non-scripted
-						environments in Top-level Content Documents can result in EPUB Publications being
-						unreadable.</p>
+					<section id="sec-scripted-spine">
+						<h5>Spine-Level Scripts</h5>
+
+						<p>A <dfn>spine-level script</dfn> is an instance of the [[!HTML]] <a
+								href="https://html.spec.whatwg.org/multipage/scripting.html#the-script-element"
+									><code>script</code></a> or [[!SVG]] <a
+								href="https://www.w3.org/TR/SVG/interact.html#ScriptElement"><code>script</code></a>
+							element contained in a <a>Top-level Content Document</a>.</p>
+
+						<p>Authors should note that <a
+								href="https://www.w3.org/TR/epub-rs-33/#sec-scripted-content-rs-reqs">support for
+								spine-level scripting in Reading Systems</a> is only recommended in fixed layout
+							documents and reflowable documents when they are set to scroll [[EPUB-RS-33]]. Furthermore,
+							Reading System support in all other contexts is optional.</p>
+
```

Comment:
I wonder whether a more direct reference to [§4.4.1](https://w3c.github.io/epub-specs/epub33/rs/#sec-scripted-content-rs-reqs) in the RS document, that is much more precise about by the properties that must be used. Ie, statements like:

> reflowable EPUB Content Documents that use the "scrolled-doc" or "scrolled-continuous" [EPUB-33] presentation modes defined by the rendition:flow property.

are much more concrete and enforceable, ie, the author should use those constraints.

That being said, I am a bit bothered by this restriction (I know, that may be an inherited feature that we cannot change). For example, a script that puts a line number to each line in a `<pre>` element does not fall into this category if the content is displayed in paginated mode, although that may make a lot of sense. I understand that the danger is to interfere with the pagination algorithm of the RS, but I still wonder whether we can be, somehow, more liberal...

Reply:
> I understand that the danger is to interfere with the pagination algorithm of the RS, but I still wonder whether we can be, somehow, more liberal...

Again, this is probably one for a new issue.

Reply:
True for the last remark, and maybe we should simply ignore this for now...